tests/onramps/test_hag_pim.py

```diff
@@ -0,0 +1,222 @@
+import pytest
+import numpy as np
+import xarray as xr
+from rasterio.transform import from_origin
+
+from fastfuels_core.onramps.hag_pim import (
+    check_same_crs,
+    check_projected_crs,
+    check_resolution,
+    resample_raster,
+    convert_to_cover,
+    interpolate_hag,
+    interpolate_pim,
+    combine_hag_pim,
+    create_gdf
+)
+
+def create_toy_raster(bounds = (-12,0,0,8),
+                      crs="EPSG:5070",
+                      nodata_value = 0,
+                      res = (1,-1),
+                      data = None):
+    x_min, x_max, y_min, y_max = bounds
+   
+    width = int((x_max - x_min) / abs(res[0]))
+    height = int((y_max - y_min) / abs(res[1]))
+    transform = from_origin(x_min, y_max, abs(res[0]), abs(res[1]))
+    if data is None:
+        data = np.ones((height, width))
+    toy_raster = xr.DataArray(
+        data,
+        dims=["y", "x"],
+        coords={
+            "x": np.linspace(x_min + res[0]/2, 
+                             x_max - res[0]/2, width),
+            "y": np.linspace(y_min - res[1]/2, 
+                             y_max + res[1]/2, height)[::-1],
+        },
+        name="toy_data")
+    toy_raster = toy_raster.rio.write_crs(crs)
+    toy_raster = toy_raster.rio.write_transform(transform)
+    toy_raster.rio.write_nodata(nodata_value, inplace=True)
+    return toy_raster
+
+@pytest.fixture
+def hag_4326():
+    data = np.array([
+        [0, 0, 0, 1, 1, 0],
+        [0, 0, 0, 0, 0, 1],
+        [0, 1, 1, 1, 1, 1],
+        [0, 1, 0, 1, 1, 1]
```

Comment:
It would be good to diversify this data instead of all edge cases. Maybe have some 0s, 1s, and values from 1-60?